src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -669,12 +655,23 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     // Return Value:
     // - the key chord that executes the given action
     // - nullptr if the action is not bound to a key chord
-    Control::KeyChord ActionMap::GetKeyBindingForAction(const winrt::hstring& cmdID) const
+    Control::KeyChord ActionMap::GetKeyBindingForAction(const winrt::hstring& cmdID)
     {
-        // Check our internal state.
-        if (const auto cmd{ _GetActionByID(cmdID) })
+        if (!_ResolvedKeyActionMapCache)
         {
-            return cmd.Keys();
+            _RefreshKeyBindingCaches();

```

Comment:
fwiw, if these collections like _ResolvedKeyActionMapCache are NOT USED on the WinRT or IDL surface of this API, _they do not need to be winrt collection types_. They will be smaller and faster.

This one, for example, sounds like an std::unordered_map from keychord to command. Or, if we enumerate the entire thing, an std::unordered_set.